uservice_ccutter/plugins/lsst_technote_bootstrap.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+"""Field-calculating plugins for lsst-technote-bootstrap.  For each
+field you want substituted, make a function whose name the field name
+you want to change and which takes a dictionary as input.  It should
+returns the value that field should have after substitution.  Then
+call the plugins substitute() function with the cookiecutter template
+type as the first string argument and your
+dictionary-requiring-substitution as input, and it will change the
+values in that dictionary.
+
+"""
+import github3
+from apikit import BackendError
+from .generic import current_year
+
+ORGHASH = {"sqr": "lsst-sqre",
+           "dmtn": "lsst-dm",
+           "smtn": "lsst-sims",
+           "ldm": "lsst"}
+
+
+def serial_number(auth, inputdict):
+    """Find the next available serial number for the specified series."""
+    series = inputdict["series"].lower()
+    # Requires that the series pick list has already been replaced with a
+    #  string.
+    # Derive github_org from the series
+    gh_org = ORGHASH[series]
+
+    # scanning the product list won't work.  We need to find the next
+    #  Github repo to use.
+
+    ghub = github3.login(auth["username"], token=auth["password"])
+    try:
+        ghub.me()
+    except github3.exceptions.AuthenticationFailed:
+        raise BackendError(status_code=401,
+                           reason="Bad credentials",
+                           content="Github login failed.")
+    max_serial = 0
+    matchstr = gh_org + "/" + series + "-"
+    for repo in ghub.repositories():
+        rnm = str(repo)
+        if rnm.startswith(matchstr):
+            serstr = rnm[(len(matchstr)):]
+            try:
+                sernum = int(serstr)
+                if sernum > max_serial:
+                    max_serial = sernum
+            except ValueError:
+                # We take "couldn't decode" as "not a serial"
+                pass
+    serial = "%03d" % (max_serial + 1)
+    # Actually the same as github_namespace, but the Jinja2 substitution will
+    #  not have happened yet.
+    if "github_repo" not in inputdict or not inputdict["github_repo"]:
+        inputdict["github_repo"] = gh_org + "/" + series + "-" + serial
+    return serial
+
+
+def github_org(auth, inputdict):
+    """Derive Github Org from the series."""
+    auth = auth
```

Comment:
This is why I am continuing to use pylint:

```(cdk2) adam@ixitxachitl:~/git/sqre-uservice-ccutter/uservice_ccutter$ pyflakes server.py
(cdk2) adam@ixitxachitl:~/git/sqre-uservice-ccutter/uservice_ccutter$ flake8 server.py
(cdk2) adam@ixitxachitl:~/git/sqre-uservice-ccutter/uservice_ccutter$ pylint server.py
No config file found, using default configuration
************* Module uservice_ccutter.server
I: 49, 0: Locally disabling unused-variable (W0612) (locally-disabled)
I: 57, 0: Locally disabling unused-variable (W0612) (locally-disabled)
I: 64, 0: Locally disabling unused-variable (W0612) (locally-disabled)
I: 74, 0: Locally disabling unused-variable (W0612) (locally-disabled)
I:104, 0: Locally disabling too-many-locals (R0914) (locally-disabled)
I:246, 0: Locally disabling too-many-locals (R0914) (locally-disabled)
W: 29,33: Unused argument 'debug' (unused-argument) 
```

Neither pyflakes nor flake8 caught the unused argument there.

I am OK with disabling pylint warnings rather than doing, say, pointless variable assignments to shut it up.  And I am 100% OK with ensuring that my code is clean under pyflakes and flake8 as well as pylint.  But I'm not OK with not using it, since it catches legitimate bugs that the other two don't.  It's fine with me if this remains an idiosyncratic choice of mine, but I find it a helpful tool for me.

I've added flymake/pyflakes to Emacs; that plus py-autopep8 seem to do a pretty good job of getting the clode syntactical and clean.